src/torchjd/autojac/mtl_backward.py

```diff
@@ -23,12 +25,92 @@
 )
 
 
+def _get_tensors_memory_id(tensor_list: list[Tensor]) -> set[int]:
+    """
+    Based on a list of lists of tensor objects returns the unique memory adress of all
+    :param tensor_list: list containing pytorch Tensor objects
+    """
+    return set([id(tensor) for tensor in tensor_list])
+
+
+def _traverse_ag_graph(grad_graph: Node) -> list[Tensor]:
```

Comment:
I think this is supposed to be a set, unless we want to allow order and repetition?

Reply:
agree done